### PR TITLE
Runtime/autonomous

### PR DESCRIPTION
**Pull request by yizow**
Depends on other pull requests.

### CODE REVIEW

**Review by nikitakit — CHANGES_REQUESTED** [review]
So what I'm seeing here is the beginnings of a custom scheduler implementation. That raises two questions:

1. Why do we need a scheduler?
2. Why are we writing one ourselves?

With respect to point 1, I would like to see a sketch of the intended student API before implementation begins. Important questions include: what is being enqueued (hibike commands, or arbitrary functions including `lambda` expressions)? What happens when something is enqueued from inside `loop` (i.e. at a rate of 30Hz)? For these and other questions, see our slack channel.

Once we have an idea of what API we want to provide, we can then address point 2 -- i.e. determine what parts we need to implement ourselves, and what parts can reuse existing components. For example, python 3.5 has a built-in `asyncio` module (plus `async`/`await` keywords), which seems like a more promising approach then writing our own event loop/scheduler.

In summary, I find it very likely that the approach in this PR will have to be significantly modified after additional discussion and prototyping take place. Therefore I don't think it makes sense to merge it at this stage.
